libraries/AP_OpticalFlow/AP_OpticalFlow_Linux.cpp

```diff
@@ -101,21 +95,21 @@ bool AP_OpticalFlow_Linux::read(optical_flow_s* report)
     i2c_integral_frame f_integral;
 
     // Perform the writing and reading in a single command
-    if (PX4FLOW_REG == 0x00) {
-        if (hal.i2c->readRegisters(PX4FLOW_I2C_ADDRESS, 0, I2C_FRAME_SIZE + I2C_INTEGRAL_FRAME_SIZE, val) != 0) {
+    if (HAL_OPTFLOW_PX4FLOW_REGISTER == 0x00) {
+        if (!_dev->read_registers(0, val, I2C_FRAME_SIZE + I2C_INTEGRAL_FRAME_SIZE)) {
             goto fail_transfer;
         }
         memcpy(&f_integral, &(val[I2C_FRAME_SIZE]), I2C_INTEGRAL_FRAME_SIZE);
     }
```

Comment:
Humn... not related to this patch, but the condition above seems to be dead code?